src/util/hass-translation.html

```diff
@@ -0,0 +1,103 @@
+<link rel='import' href='../../build-temp/translationMetadata.html' />
+
+<script>
+function getActiveTranslation() {
+  // Perform case-insenstive comparison since browser isn't required to
+  // report languages with specific cases.
+  const lookup = {};
+  /* eslint-disable no-undef */
+  Object.keys(window.translationMetadata).forEach(function (tr) {
+    lookup[tr.toLowerCase()] = tr;
+  });
+
+  // Search for a matching translation from most specific to general
+  function languageGetTranslation(language) {
+    const subtags = language.toLowerCase().split('-');
+
+    for (var i = subtags.length; i >= 1; i--) {
+      const lang = subtags.slice(0, i).join('-');
+
+      if (lookup[lang]) {
+        return lookup[lang];
+      }
+    }
+    return null;
+  }
+
+  var translation = null;
+  if (window.localStorage.selectedLanguage) {
+    translation = languageGetTranslation(JSON.parse(window.localStorage.selectedLanguage));
+    if (translation) {
+      return translation;
+    }
+  } else if (navigator.languages) {
+    for (var i = 0; i < navigator.languages.length; i++) {
+      translation = languageGetTranslation(navigator.languages[i]);
+      if (translation) {
+        return translation;
+      }
+    }
+  } else {
+    translation = languageGetTranslation(navigator.language || navigator.userLanguage);
+    if (translation) {
+      return translation;
+    }
+  }
+
+  // Final fallback
+  return 'en';
+}
+
+// Store loaded translations in memory so translations are available immediately
+// when DOM is created in Polymer. Even a cache lookup creates noticable latency.
+const translations = {};
+
+window.getTranslation = function (translationInput) {
+  var translation = translationInput || getActiveTranslation();
+  /* eslint-disable no-undef */
+  var translationFingerprint = window.translationMetadata[translation].fingerprint;
+
+  // Create a promise to fetch translation from the server
+  if (!translations[translationFingerprint]) {
+    translations[translationFingerprint] = new Promise(function (resolve) {
+      var translationUrl = '/static/translations/' + translationFingerprint;
+      var request = new XMLHttpRequest();
```

Comment:
Let's use [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) ? It's only not supported in [IE11](http://caniuse.com/#search=fetch) and can be added to our polyfill if we want.

Reply:
Was going to add this, but it seems that [IE11 doesn't support promises anyway](http://caniuse.com/#search=Promise), which would mean we've already dropped support for them, unless I'm missing something.